File: server/matches/procon/agent.py
# -*- coding: utf-8 -*-
# Agentクラス
class Agent:
    # コンストラクタ
    def __init__(self, agentID, x, y):
        self.agentID = agentID
        self.x = x
        self.y = y
        self.action_type = "stay"
        self.dx = 0
        self.dy = 0
        self.apply = 1

    # 転置
    def transpose(self):
        self.x, self.y = self.y, self.x

    # 試合状態取得APIのレスポンスの一部, 行動結果の送信用辞書の作成
    def create_result_dictionary(self, turn):
        action = {}
        action["actionID"] = self.agentID
        action["dx"] = self.dx
        action["dy"] = self.dy
        action["type"] = self.action_type
        action["apply"] = self.apply
        action["turn"] = turn
        return action

File: solver/solver/game/cpp/agent.cpp
#include "game/Agent.hpp"

Agent::Agent() {}

Agent::Agent(int id_) { agent_id = id_; }

Agent::Agent(int id_, int x_, int y_) {
    agent_id = id_;
    x = x_;
    y = y_;
}

void Agent::set_action(int dx_, int dy_, bool will_flip) {
    dx = dx_;
    dy = dy_;
    if (will_flip == true) action_type = "remove";
    if (will_flip == false) action_type = "move";
    if (dx == 0 and dy == 0) action_type = "stay";
}